ui/components/multichain/pages/review-permissions-page/review-permissions-page.tsx

```diff
@@ -145,28 +165,84 @@ export const ReviewPermissions = () => {
     setShowNetworkToast(true);
   };
 
-  const accounts = useSelector(getUpdatedAndSortedAccounts);
-  const evmAccounts: MergedInternalAccount[] = useMemo(() => {
-    return accounts.filter((account: InternalAccount) =>
-      isEvmAccountType(account.type),
-    );
-  }, [accounts]);
+  const allAccounts = useSelector(
+    getUpdatedAndSortedAccountsWithCaipAccountId,
+  ) as MergedInternalAccountWithCaipAccountId[];
 
-  const connectedAccountAddresses = useSelector((state) =>
-    getPermittedAccountsForSelectedTab(state, activeTabOrigin),
-  ) as string[];
+  const nonRemappedConnectedAccountAddresses = useSelector((state) =>
+    getAllPermittedAccountsForSelectedTab(state, activeTabOrigin),
+  ) as CaipAccountId[];
 
-  const handleSelectAccountAddresses = (addresses: string[]) => {
-    if (addresses.length === 0) {
+  // This remaps EVM caip account addresses to match the 'eip155:0'
+  // value that is currently set in InternalAccount.scopes[0] for
+  // EOA EVM accounts. This logic will need to be updated to
+  // support non EOA accounts.
+  const connectedAccountAddresses = uniq(
+    nonRemappedConnectedAccountAddresses.map((caipAccountId) => {
+      const {
+        address,
+        chain: { namespace },
+      } = parseCaipAccountId(caipAccountId);
+      if (namespace === KnownCaipNamespace.Eip155) {
+        // this is very hacky, but it works for now
+        return `eip155:0:${address}` as CaipAccountId;
+      }
+      return caipAccountId;
+    }),
+  );
+
+  const handleSelectAccountAddresses = (caipAccountIds: CaipAccountId[]) => {
+    if (caipAccountIds.length === 0) {
       setShowDisconnectAllModal(true);
       return;
     }
 
+    // Temporarily pass only the address part of the caipAccountId until
+    // the rest of the UI has been refactored to work with CaipAccountIds
+    const addresses = caipAccountIds.map((caipAccountId) => {
+      const { address } = parseCaipAccountId(caipAccountId);
```

Comment:
We are reparsing the account IDs a bunch of times here (in the next couple blocks), perhaps we can reuse a list of parsed IDs instead?

Reply:
improved here: https://github.com/MetaMask/metamask-extension/pull/29887/commits/1c781661e2290b70905eb4345e1df99595bc2dfd